### PR TITLE
Fix localized download text

### PR DESCRIPTION
**Pull request by hunterwerlla**
- Fix some more gratuitous title case
- Set default file name as log stream name
- Set CloudWatch Download text properly
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: resources/resources/software/aws/toolkits/resources/localized_messages.properties [hunk 1]
```diff
@@ -598,7 +598,8 @@ cloud_debug.instrument.production_warning.checkbox_label=I confirm {0} is not a
 cloud_debug.instrument.production_warning.title=Warning!
 lambda.run_configuration.debug_host=Debug Host:
 lambda.sam.debugHost.tooltip=The host to connect the debugger to
-cloudwatch.logs.save_action=Save To A File
+cloudwatch.logs.save_action=Save to a File
+cloudwatch.logs.download=Download
```
Comment (abrooksv) [inline resources/resources/software/aws/toolkits/resources/localized_messages.properties]: I would put a better description like we did for S3.

Also what happens if they select a folder

Comment (hunterwerlla) [reply resources/resources/software/aws/toolkits/resources/localized_messages.properties]: Actually, I don't think this text is used lol

If they select a folder it is saved into that folder as expected

Comment (abrooksv) [reply resources/resources/software/aws/toolkits/resources/localized_messages.properties]: It is used in the non-native browser.

Comment (abrooksv) [reply resources/resources/software/aws/toolkits/resources/localized_messages.properties]: You can force it by calling setForcedToUseIdeaFileChooser(true) on the FileSaverDescriptor to see what it looks like

Comment (hunterwerlla) [reply resources/resources/software/aws/toolkits/resources/localized_messages.properties]: Cool, updated the text to be more descriptive

**Pull request merged**